### DISCUSSION TITLE
filter result based on user input

### DISCUSSION THREAD

**Discussion by itsmefarhan** [knoxpo/dart_algolia#126] [Q&A]
in algolia I have an index "users" which contains username and email fields. I would like to filter the result by username. I think I am missing something here because my query doesnt work.

AlgoliaQuery query = algolia.instance.index('users').query('test'); // search word test
query = query.facetFilter('username:test'); // only where username is test

final data = await query.getObjects();
print(data.hits);
it prints empty list. Can anyone tell me how to fix it? Thanks

**Comment by muntasir-ifaz574** [comment]
AlgoliaQuery query = algolia.instance.index('users').query('test');
query = query.filters('username:test');

final data = await query.getObjects();
print(data.hits);

You need to use the filters method instead of facetFilter.
